web/src/enterprise/campaigns/list/CampaignNode.tsx

```diff
@@ -5,31 +5,62 @@ import { renderMarkdown } from '../../../../../shared/src/util/markdown'
 import { CampaignsIcon } from '../icons'
 import { Link } from '../../../../../shared/src/components/Link'
 import classNames from 'classnames'
+import { changesetStateIcons, changesetStatusColorClasses } from '../detail/changesets/presentation'
+import formatDistance from 'date-fns/formatDistance'
+import parseISO from 'date-fns/parseISO'
 
 interface Props {
-    node: GQL.ICampaign
+    node: Pick<
+        GQL.ICampaign,
+        'id' | 'plan' | 'closedAt' | 'name' | 'description' | 'changesets' | 'changesetPlans' | 'createdAt'
+    >
+    /* Used for testing purposes. Sets the current date */
+    now?: Date
 }
 
 /**
  * An item in the list of campaigns.
  */
-export const CampaignNode: React.FunctionComponent<Props> = ({ node }) => {
+export const CampaignNode: React.FunctionComponent<Props> = ({ node, now = new Date() }) => {
     const campaignIconClass = node.closedAt ? 'text-danger' : 'text-success'
+    const OpenChangesetIcon = changesetStateIcons[GQL.ChangesetState.OPEN]
+    const MergedChangesetIcon = changesetStateIcons[GQL.ChangesetState.MERGED]
+    const changesetCountByState = (state: GQL.ChangesetState): number =>
+        node.changesets.nodes.reduce((prev, next) => prev + (next.state === state ? 1 : 0), 0)
     return (
         <li className="card p-2 mt-2">
-            <div className="d-flex">
-                <CampaignsIcon className={classNames('icon-inline mr-2 flex-shrink-0', campaignIconClass)} />
+            <div className="d-flex align-items-center">
+                <CampaignsIcon
+                    className={classNames('icon-inline mr-2 flex-shrink-0 align-self-stretch', campaignIconClass)}
+                    data-tooltip={node.closedAt ? 'Closed' : 'Open'}
+                />
                 <div className="campaign-node__content">
-                    <h3 className="m-0">
-                        <Link to={`/campaigns/${node.id}`} className="d-flex align-items-center text-decoration-none">
-                            {node.name}
-                        </Link>
-                    </h3>
+                    <div className="m-0 d-flex align-items-baseline">
+                        <h3 className="m-0 d-inline-block">
+                            <Link to={`/campaigns/${node.id}`}>{node.name}</Link>
+                        </h3>
+                        <span className="badge badge-light ml-2">{node.plan?.type ?? 'manual'}</span>
+                        <small className="ml-2 text-muted" data-tooltip={node.createdAt}>
+                            {formatDistance(parseISO(node.createdAt), now)} ago
+                        </small>
+                    </div>
                     <Markdown
-                        className="text-truncate"
-                        dangerousInnerHTML={renderMarkdown(node.description, { plainText: true })}
+                        className={classNames('text-truncate', ...(node.description ? [] : ['text-muted']))}
```

Comment:
`class-names` can handle conditional classes afaik (basically its core feature)
```suggestion
                        className={classNames('text-truncate', node.description && 'text-muted')}
```

Reply:
ah cute